### PR TITLE
󠀡󠀡

### PR DESCRIPTION
**Pull request by niclasdriver**
󠀡󠀡

### CODE REVIEW

File: ChaosMod/Effects/db/Misc/MiscFlashbang.cpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+		f_alpha--;
+	}
+}
+
+static void OnStop()
+{
+	f_alpha = 510;
+}
+
+static RegisterEffect registerEffect(EFFECT_MISC_FLASHBANG, nullptr, OnStop, OnTick);
```
Comment (TheRealAyCe) [inline ChaosMod/Effects/db/Misc/MiscFlashbang.cpp:27]: Why use OnStop and not OnStart to set f_alpha? Then you'd only need to define 510 once.

**Comment by ProfessorBiddle** [comment]
Looks a lot like the flash grenade effect from Hitman 1-2-3 too

File: ChaosMod/Effects/db/Misc/MiscFlashbang.cpp (hunk 1 above)
Comment (kolyaventuri) [reply ChaosMod/Effects/db/Misc/MiscFlashbang.cpp:27]: ^ +1, I would just instantiate `f_alpha` at the top level, and set it to 510 on an `OnStart`

**Comment by kolyaventuri** [comment]
This doesn't make sense to be a timed effect. I would opt to use `WAIT(0)` and just run the flashbang effect in the `OnStart`. Just use like, a `while (f_alpha > 0) { . . . }` in the `OnStart`, and decrement it there with a `WAIT(0)` at the end of the loop

**Comment by niclasdriver** [comment]
[Optional sound effects](https://www.dropbox.com/s/rmzmfaox1nfz677/Flashbang%20Sound%20Effects.zip?dl=1)

**Comment by Last0xygen** [comment]
🤔 why?

**Pull request closed without merging**